### PR TITLE
Revisiting i18n

### PR DESCRIPTION
**Pull request by augustf**
This branch represents the beginning of finalizing our i18n work. The overall work on this is being tracked in #1250

### CODE REVIEW

**Comment by simplysoft** [comment]
A lot of intended duplication for translation was removed and you are now again trying to generate human language, which tends to be easy for english, but can be hard with e.g. Japanese or impossible with even more obscure languages. Turns out that you can't easily generate sentences for all languages using a fixed structure. This is why human translators are needed.

See also our comments at https://github.com/concerto/concerto/issues/1135#issuecomment-68591222

File: app/controllers/concerto_plugins_controller.rb [hunk 1]
```diff
@@ -49,7 +49,7 @@ def create
         restarted = restart_webserver()
       end
       if restarted
-        flash[:notice] = t(:plugin_created)
+        flash[:notice] = t(:was_created, name: @concerto_plugin.name, theobj: t(:plugin))
```
Comment (simplysoft) [inline app/controllers/concerto_plugins_controller.rb:52]: Coders love to reduce duplication, but this does not work for natural languages. It looks like this duplication is a lot of unnecessary work for translators, but it makes their life actually easier. They can e.g. use Translation memory to speed up the process and still have the ability to respect some language specifics when translating. And it is much easier to translate a real sentence than something like `The %{name} %{theobj} was successfully created.` where the translator needs to account for all future possible words that might be filled in there. 

And this simple string, can't be translated to a pretty similar language to english, which happens to be German. There we would have to distinguish between male and female `%{name}`and use different articles and can't just use `The`. So in german it should be `%{article} %{name} %{theobj} was successfully created.`. 

When you are trying to translate this into even more different languages, that have complexity different grammatical structures and concepts, it is impossible to correctly translate this.  
You will end up with `%{article} %{name} %{theobj} %{temporalverb} %{adjective} %{verb}.`

**Comment by augustf** [comment]
@simplysoft I'm afraid you're right. I thought I could save this scheme by allowing essentially recursive translations, but that won't work. We presently have a situation where a substantial minority of our strings are untranslated as they've not been entered in a YML file, but at present, I don't have a solution.

**Comment by augustf** [comment]
@simplysoft Would many of the lines we have such as  ```are_you_sure_delete_model_key: Are you sure you want to delete %{model} "%{key}"?``` also be problematic with your translations?

**Comment by simplysoft** [comment]
@augustf Yes, this also would apply to ```are_you_sure_delete_model_key: Are you sure you want to delete %{model} "%{key}"?```. For german it works, but I bet there is some language out there where it would not work. 
One middle course could be to still use those composing yml strings but allowing to overwrite a default translation with specific ones. We have already used this e.g. here:

https://github.com/concerto/concerto/blob/master/app/views/layouts/_errors.html.erb
https://github.com/concerto/concerto/blob/master/config/locales/de.yml#L653-L692

This still allows to only add one generic translation (for languages where this works), but gives the flexibility to add special translations for each special case. 
(BTW: Writing out all possible cases with their full translation key is for better tool support, refactoring, translation support etc.)

**Pull request closed without merging**